client/quota.go

```diff
@@ -24,29 +24,30 @@ import (
 	"encoding/json"
 	"fmt"
 
+	"github.com/snapcore/snapd/gadget/quantity"
 	"golang.org/x/xerrors"
 )
 
 type postQuotaData struct {
-	Action    string   `json:"action"`
-	GroupName string   `json:"group-name"`
-	Parent    string   `json:"parent,omitempty"`
-	Snaps     []string `json:"snaps,omitempty"`
-	MaxMemory uint64   `json:"max-memory,omitempty"`
+	Action      string            `json:"action"`
+	GroupName   string            `json:"group-name"`
+	Parent      string            `json:"parent,omitempty"`
+	Snaps       []string          `json:"snaps,omitempty"`
+	Constraints map[string]string `json:"constraints,omitempty"`
```

Comment:
other bits for our API use int64 or similar for sizes, so using strings here is quite inconsistent, either we go int64 or interface{} ?

Reply:
mmm I don't particularly like using interface{}, but maybe there's an easy way to make this work in a clearer manner, I will try a few things out

Reply:
well I went with interface{} it was a bit of work and maybe is too much, but I was trying to keep it genericly a map so that we can more easily format things in the output of `snap quota[s]`, but perhaps that would be too easy and we need something silly and duplicative like

```go
type foo struct {
    Snaps []string `json:"snaps,omitempty"`
    Constraints struct {
        Memory quantity.Size `json:"memory,omitempty"`
    } `json:"constraints,omitempty"`
    Current struct {
        Memory quantity.Size `json:"memory,omitempty"`
    } `json:"current,omitempty"`
}
```

Reply:
we could introduce a struct QuotaValues ? (finding a good name is a bit annoying)

Reply:
what is your preference?

Reply:
I think introducing a struct, at least at the level of the client package, which explicitly lists what we support currently is my preference, because it's the most self-documenting approach. though it poses a little naming problem

Reply:
ok, sure I will add a QuotaValues to client and use that instead

Reply:
struct added